core/src/sumeragi/main_loop.rs

```diff
@@ -628,6 +657,9 @@ fn handle_message<F: FaultInjection>(
                 let voting_block_hash = voted_block.block.hash();
 
                 if hash == voting_block_hash.transmute() {
+                    // The manipulation of the topology relies upon all peers seeing the same signature set.
+                    // Therefore we must clear the signatures and accept what the proxy tail giveth.
+                    voted_block.block.signatures.clear();
                     add_signatures::<true>(&mut voted_block, signatures.transmute());
```

Comment:
I don't like the `clear` being introduced

```suggestion
                    voted_block.block.signatures = signatures;
```

what would happen in this case?

Reply:
That would create a security issue. `add_signatures` contains validation code that must run.

Reply:
This is not good because you're breaking `SignaturesOf` invariants. Check docs on `SignaturesOf`

Reply:
`SignaturesOf` is worth encapsulating. That said, we don't even know off the top of our heads what invariants need to be upheld, and that's a problem

Reply:
Good point. I've now updated the docs to remove that invariant.